Code/Editor/CryEditDoc.cpp

```diff
@@ -1504,36 +1498,33 @@ bool CCryEditDoc::LoadXmlArchiveArray(TDocMultiArchive& arrXmlAr, const QString&
 {
     auto pIPak = GetIEditor()->GetSystem()->GetIPak();
 
-    //if (m_pSWDoc->IsNull())
+    CXmlArchive* pXmlAr = new CXmlArchive();
+    if (!pXmlAr)
```

Comment:
I know this is legacy cry code, but this seems unnecessary of a check. Or even, why did they dynamically create this object only to reference its dereferenced instance. This also looks like a memory leak as well.

Reply:
Agreed. I'll try getting this to pass AR as is, but then this definitely warrants another pass.